client/txn.go

```diff
@@ -90,6 +90,11 @@ func newTxn(db DB, depth int) *Txn {
 	return txn
 }
 
+// GetState returns the transaction protobuf.
+func (txn *Txn) GetState() proto.Transaction {
```

Comment:
We don't need it, as you observed `haveTxnWrite` is only relevant for requests which are queued at the client (i.e. in a batch), but in this scenario we store the state when nothing is pending, so it's always `false`. Same for `haveEndTxn`, it's also going to be `false` unless you ended the transaction, in which case the `State` is of no more importance.